lib/rucio/core/replica.py

```diff
@@ -303,10 +304,15 @@ def __declare_bad_file_replicas(pfns, rse_id, reason, issuer, status=BadFilesSta
         replicas_list.append((scope, name, path))
 
     for scope, name, path, __exists, already_declared, size in __exist_replicas(rse_id=rse_id, replicas=replicas_list, session=session):
-        if __exists and ((status == BadFilesStatus.BAD and not already_declared) or status == BadFilesStatus.SUSPICIOUS):
+        if __exists and \
+                (
+                    status == BadFilesStatus.BAD and (force or not already_declared)
+                    or status == BadFilesStatus.SUSPICIOUS
+                ):
             declared_replicas.append({'scope': scope, 'name': name, 'rse_id': rse_id, 'state': ReplicaState.BAD})
-            new_bad_replica = models.BadReplicas(scope=scope, name=name, rse_id=rse_id, reason=reason, state=status, account=issuer, bytes=size)
-            new_bad_replica.save(session=session, flush=False)
+            if not already_declared:
```

Comment:
There's a change in the logic here. Before, we always inserted a new row if `status == BadFilesStatus.SUSPICIOUS`. The more a replica got SUSPICIOUS counts, the more likely it was bad. This is used by the `replica-recover` to apply automatic recovery policy. Can you please keep the old behaviour ?

Reply:
Thanks Cedric. Now I understand that logic. I think I restored it.